samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -135,6 +139,8 @@ public RemoteTableDescriptor(String tableId) {
     Preconditions.checkNotNull(readFn, "null read function");
     Preconditions.checkNotNull(retryPolicy, "null retry policy");
     this.readRetryPolicy = retryPolicy;
+    this.readRetryPolicy.setTableId(String.format("%s.%s", tableId,
```

Comment:
Another way could be: when TableRetryPolicy implements toConfig method, it doesn't put the tableId as part of the key. We then append the tableId to the key here in RemoteTableDescriptor`.
This way, a TableRetryFunction can be shared by any number of tables, because there is no tableId associated to it.